### PR TITLE
Update interface and monitoring

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

**Review by MykolaSolyanko — APPROVED** [review]

File: include/aos/common/monitoring/average.hpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+     *
+     * @param data monitoring data.
+     * @return Error.
+     */
+    Error Update(const NodeMonitoringData& data);
+
+    /**
+     * Returns average data.
+     *
+     * @param data monitoring data.
```
Comment (mlohvynenko) [inline include/aos/common/monitoring/average.hpp:41]:
```suggestion
     * @param[out] data monitoring data.
```

File: include/aos/common/monitoring/monitoring.hpp [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    /**
+     * Destructor
+     */
+    virtual ~ResourceUsageProviderItf() = default;
+
+    /**
+     * Returns node monitoring data.
+     *
+     * @param nodeID node ident.
+     * @param monitoringData monitoring data.
```
Comment (mlohvynenko) [inline include/aos/common/monitoring/monitoring.hpp:146]:
```suggestion
     * @param[out] monitoringData monitoring data.
```

File: include/aos/common/monitoring/monitoring.hpp [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+     * @return Error.
+     */
+    virtual Error GetNodeMonitoringData(const String& nodeID, MonitoringData& monitoringData) = 0;
+
+    /**
+     * Returns instance monitoring data.
+     *
+     * @param instanceID instance ID.
+     * @param monitoringData monitoring data.
```
Comment (mlohvynenko) [inline include/aos/common/monitoring/monitoring.hpp:155]:
```suggestion
     * @param[out] monitoringData monitoring data.
```

File: include/aos/sm/instance.hpp [hunk 4]
```diff
@@ -83,17 +83,17 @@ class Instance {
...
-    uint64_t AosVersion() const
+    StaticString<cVersionLen> GetServiceVersion() const
     {
         if (mService) {
-            return mService->Data().mVersionInfo.mAosVersion;
+            return mService->Data().mVersion;
         }
 
-        return 0;
+        return "";
```
Comment (mlohvynenko) [inline include/aos/sm/instance.hpp:96]: should we return some default semver in this case, like "0.0.0" ?

File: src/common/monitoring/resourcemonitor.cpp [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+    mResourceUsageProvider = &resourceUsageProvider;
+    mMonitorSender         = &monitorSender;
+    mConnectionPublisher   = &connectionPublisher;
+
+    NodeInfo nodeInfo;
+
+    auto err = nodeInfoProvider.GetNodeInfo(nodeInfo);
+    if (!err.IsNone()) {
+        return err;
```
Comment (mlohvynenko) [inline src/common/monitoring/resourcemonitor.cpp:33]:
```suggestion
        return AOS_ERROR_WRAP(err);
```

File: include/aos/common/monitoring/monitoring.hpp [hunk 6]
```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
```
Comment (mlohvynenko) [inline include/aos/common/monitoring/monitoring.hpp:2]:
```suggestion
 * Copyright (C) 2024 Renesas Electronics Corporation.
```

File: src/common/monitoring/resourcemonitor.cpp [hunk 7]
```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (C) 2023 Renesas Electronics Corporation.
```
Comment (mlohvynenko) [inline src/common/monitoring/resourcemonitor.cpp:2]:
```suggestion
 * Copyright (C) 2024 Renesas Electronics Corporation.
```

**Review by mlohvynenko — APPROVED** [review]

File: include/aos/sm/instance.hpp (hunk 4 above)
Comment (al1img) [reply include/aos/sm/instance.hpp:96]: Interesting question. Let's return empty string for now. Then will see.

File: include/aos/common/monitoring/monitoring.hpp (hunk 6 above)
Comment (al1img) [reply include/aos/common/monitoring/monitoring.hpp:2]: These files moved not created.

File: src/common/monitoring/resourcemonitor.cpp (hunk 7 above)
Comment (al1img) [reply src/common/monitoring/resourcemonitor.cpp:2]: The same.

**Review by mlohvynenko — APPROVED** [review]

File: src/common/monitoring/resourcemonitor.cpp [hunk 8]
```diff
@@ -105,11 +97,40 @@ Error ResourceMonitor::StopInstanceMonitoring(const String& instanceID)
...
 
-    return mNodeMonitoringData.mServiceInstances
-        .Remove([&instanceID](const InstanceMonitoringData& instance) { return instance.mInstanceID == instanceID; })
-        .mError;
+    auto nodeError = mNodeMonitoringData.mServiceInstances
+                         .Remove([&instanceID](const InstanceMonitoringData& instance) {
+                             return instance.mInstanceID == instanceID;
+                         })
+                         .mError;
+    if (!nodeError.IsNone() && err.IsNone()) {
```
Comment (mykola-kobets-epam) [inline src/common/monitoring/resourcemonitor.cpp]: it seems err is always eNone

**Review by mykola-kobets-epam — APPROVED** [review]

**Pull request merged**